### PR TITLE
Completed chitter challenge

### PR DESCRIPTION
**Pull request by micosmin**
Implemented all user stories
More testing is needed
Refactoring is needed to lower dependency
Functionality to reply not implemented

### CODE REVIEW

**Comment by micosmin** [comment]
Plenty of travis ci tests, most of them because I missed that Sinatra::Flash was registered in :development mode. I want to keep them to remind me of how not to do this:)

Some of them are due to testing rake db:create and migrate, which while it worked locally, on Travis it had some issues

Also, some are because of the sending of an email functionality, which does not work from Travis in this format, as my credentials are stored locally - Working on mocking the email class.

All tests pass now

File: APPROACH.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+Step 1
```
Comment (jpalmerr) [inline APPROACH.md]: Really impressive addition to readme

File: app/chitter.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    session[:user_id] = user.id
+    redirect '/peeps'
+  end
+
+  get '/sessions/new' do
+    redirect '/'
+  end
+
+  post '/sessions/new' do
+    params['type'] == 'login' ? erb(:'sessions/login') : erb(:'sessions/signup')
```
Comment (jpalmerr) [inline app/chitter.rb]: efficient conditional

File: spec/feature/add_tag_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# TO DO: hitting send won't work on Travis as credentails are hidden
+
+feature 'Add tag' do
+
+  scenario 'User can tag another user in a peep' do 
+    User.create(name: 'Momo', username: 'momo', email: 'momo@gmail.com', password: 'fefwefwe')
```
Comment (jpalmerr) [inline spec/feature/add_tag_spec.rb]: Could use doubles/mocks

**Pull request closed without merging**